third_party/terraform/resources/resource_project_service_identity.go.erb

```diff
@@ -63,21 +67,31 @@ func resourceProjectServiceIdentityCreate(d *schema.ResourceData, meta interface
 		return fmt.Errorf("Error creating Service Identity: %s", err)
 	}
 
-	err = serviceUsageOperationWaitTime(
-		config, res, project, "Creating Service Identity",
+	var opRes map[string]interface{}
+	err = serviceUsageOperationWaitTimeWithResponse(
+		config, res, &opRes, project, "Creating Service Identity",
 		d.Timeout(schema.TimeoutCreate))
-
 	if err != nil {
 		return err
 	}
 
+	log.Printf("[DEBUG] Finished creating Service Identity %q: %#v", d.Id(), res)
+
 	id, err := replaceVars(d, config, "projects/{{project}}/services/{{service}}")
 	if err != nil {
 		return fmt.Errorf("Error constructing id: %s", err)
 	}
 	d.SetId(id)
 
-	log.Printf("[DEBUG] Finished creating Service Identity %q: %#v", d.Id(), res)
+	emailVal, ok := opRes["email"]
+	if !ok {
```

Comment:
If the response doesn't contain an email should we fail during the create method? Is returning the generated email a requirement of this request succeeding?

Reply:
Based on the API I believe we should always get an email  in the response. @xdqyzy can you please confirm?

The "right" way to implement this would be to call a READ endpoint, but there is none for this API.